packages/pnpm/test/recursive/misc.ts

```diff
@@ -360,6 +361,53 @@ test('recursive installation of packages with hooks', async t => {
   t.ok(lockfile2.packages['/dep-of-pkg-with-1-dep/100.1.0'])
 })
 
+test('recursive installation of packages in workspace ignores hooks in package', async t => {
+  // This test hangs on Appveyor for some reason
+  if (isCI && isWindows()) return
+  preparePackages(t, [
+    {
+      name: 'project-1',
+      version: '1.0.0',
+
+      dependencies: {
+        'is-positive': '1.0.0',
+      },
+    },
+    {
+      name: 'project-2',
+      version: '1.0.0',
+
+      dependencies: {
+        'is-negative': '1.0.0',
+      },
+    },
+  ])
+
+  process.chdir('project-1')
+  const pnpmfile = `
+    module.exports = { hooks: { readPackage } }
+    function readPackage (pkg) {
+      pkg.dependencies = pkg.dependencies || {}
+      pkg.dependencies['dep-of-pkg-with-1-dep'] = '100.1.0'
+      return pkg
+    }
+  `
+  await fs.writeFile('pnpmfile.js', pnpmfile, 'utf8')
+
+  process.chdir('../project-2')
+  await fs.writeFile('pnpmfile.js', pnpmfile, 'utf8')
```

Comment:
why don't you also add a pnpmfile in the root of the workspace to see that is ignored as well?

Reply:
Not 100% sure what you meant by this; tried to clarify the tests to more clearly test the behavior.